File: hackerrank/easy/0021-utopian-tree/swift/solution.swift
// solution.swift

func utopianTree(n: Int) -> Int {
    var h = 1
    if n == 0 { return h }
    for cycle in 1...n {
        if cycle % 2 == 1 { // spring (odd)
            h *= 2
        } else {            // summer (even)
            h += 1
        }
    }
    return h
}

// --- demo run ---
print(utopianTree(n: 0))  // 1
print(utopianTree(n: 1))  // 2
print(utopianTree(n: 4))  // 7
print(utopianTree(n: 5))  // 14

File: hackerrank/easy/0021-utopian-tree/js/solution.mjs
// solution.mjs

export function utopianTree(n) {
  let h = 1;
  for (let cycle = 1; cycle <= n; cycle++) {
    if (cycle % 2 === 1) h *= 2; // spring
    else h += 1;                 // summer
  }
  return h;
}

// --- demo run ---
console.log(utopianTree(0)); // 1
console.log(utopianTree(1)); // 2
console.log(utopianTree(4)); // 7
console.log(utopianTree(5)); // 14